src/fidesapi/routes/generate.py

```diff
@@ -0,0 +1,153 @@
+    """
+
+    organization_key: str
+    generate: Generate
+
+
+class GeneratedResponse(BaseModel):
+    """
+    The model to hous the response for generated infrastructure.
+    """
+
+    generate_results: Optional[List[System]]
+
+
+router = APIRouter(tags=["Generate"], prefix=f"{API_PREFIX}/generate")
+
+
+@router.post(
+    "/",
+    status_code=status.HTTP_200_OK,
+    response_model=GeneratedResponse,
+    responses={
+        status.HTTP_400_BAD_REQUEST: {
+            "content": {
+                "application/json": {
+                    "example": {"detail": "Unable to generate resources."}
+                }
+            }
+        },
+    },
+)
+async def generate(
+    generate_request_payload: GenerateRequestPayload, response: Response
+) -> Dict:
+    """
+    A multi-purpose endpoint to generate Fides resources based on existing
+    infrastructure
+
+    Currently generates Fides resources for the following:
+    * AWS: Systems
+
+    In the future, this will include options for other Systems & Datasets,
+    examples include:
+    * Okta: Systems
+    * Snowflake: Datasets
+
+    All config secrets should be encoded as a minor security precaution, using the
+    `obscure_string` function in `fidesapi.routes.util`
+
+    All production deployments should implement HTTPS for security purposes
+    """
+    organization = await get_resource(
+        sql_model_map["organization"], generate_request_payload.organization_key
+    )
+    if generate_request_payload.generate.target.lower() == "aws":
+        try:
+            import boto3  # pylint: disable=unused-import
+        except ModuleNotFoundError:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Packages not found, ensure aws is included: fidesctl[aws]",
+            )
+        log.info("Setting config for AWS")
+        generated_systems = generate_aws(
+            aws_config=AWSConfig(**generate_request_payload.generate.config.dict()),
+            organization=organization,
+        )
+    return {"generate_results": generated_systems}
+
+
+def generate_aws(
+    aws_config: AWSConfig, organization: Organization
+) -> List[Dict[str, str]]:
+    """
+    Returns a list of Systems found in AWS.
+    """
+    from botocore.exceptions import ClientError
+
+    config = {
+        "region_name": aws_config.region_name,
+        "aws_access_key_id": unobscure_string(aws_config.aws_access_key_id),
+        "aws_secret_access_key": unobscure_string(aws_config.aws_secret_access_key),
+    }
+    try:
+        log.info("Generating systems from AWS")
+        aws_systems = generate_aws_systems(organization=organization, aws_config=config)
+    except ClientError as error:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=error)
+    except:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Unable to generate resources",
+        )
```

Comment:
The default exception type should not really be a 400. If there is a use case where the client has provided an invalid request we should be explicitly finding that. We should not return a BadRequest response without telling the client what is wrong with their request. An unexpected error should map to a 5xx as it means we are failing and are unsure why.

Reply:
I see that the exception handling was addressed already above comments. I think a better place to start is to look for auth failures explicitly and only for those return our own 4xx. I think we can do this without importing any models. Anything else should be treated as a 5xx and we can build on that in the future.